client/app/services/dashboard.js

```diff
@@ -154,6 +154,24 @@ function transformResponse(data) {
   return data;
 }
 
+function transformRecentResponse(data) {
+  if (!data.results) return transformSingle(data);
+  const recentDashboards = localStorage.getItem("recentDashboards");
+  if (!recentDashboards) return { ...data, results: [] };
+  const parsedRecentDashboards = JSON.parse(recentDashboards);
+  const recentDashboardsIds = _.map(parsedRecentDashboards, dashboard => dashboard.id);
+  const allUserDashboards = _.map(data.results, transformSingle);
+  const userRecentDashboards = _.filter(allUserDashboards, dashboard => recentDashboardsIds.includes(dashboard.id.toString()));
+  const sortedUserRecentDashboards = userRecentDashboards.sort((dashboardA, dashboardB) => {
+    const priorityOfDashboardA = _.find(parsedRecentDashboards, dashboard => dashboard.id === dashboardA.id.toString()).priority;
+    const priorityOfDashboardB = _.find(parsedRecentDashboards, dashboard => dashboard.id === dashboardB.id.toString()).priority;
+    if (priorityOfDashboardA < priorityOfDashboardB) return 1;
+    if (priorityOfDashboardA > priorityOfDashboardB) return -1;
+    return 0;
+  });
+  return { ...data, results: sortedUserRecentDashboards };
```

Comment:
This "priority" thing feels a bit complex 🤔, can you elaborate a bit on what were your thoughts to order the list?

My initial thought was to have saved in the order we want and on every "track action" (view/execute or whatever we want to use to track) it: 1. Remove the existing if it is part of the list already; 2. Add it in the beginning of the list;

Reply:
So basically biggest priority implies in appearing first, every time i need to update the "recents" array, i check for the biggest value in the array, and give my new item the priority value of the biggest value + 1, that way i only need to sort the array when i'm displaying it

Reply:
@gabrieldutra Done!